### PR TITLE
Add Jumper Aion Nano target

### PR DESCRIPTION
**Pull request by norulers**
/src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp
52     int fuzz = 100  

5 direction keys can be used normally

### CODE REVIEW

**Comment by JyeSmith** [comment]
@norulers can we please arrange some samples to Devs first.  Users then know the hardware has been tested and supported.

File: src/include/target/AION_NANO_2400_TX.h
Comment (brandonrc) [inline src/include/target/AION_NANO_2400_TX.h:36]: This might be an issue if your fuzz value is 100 like you said in the comment. 

0 to 35 is your smallest difference and that divided by 2 will give us a safe fuzz value of `17.5`. We cannot use the 100 you mentioned.

**Comment by brandonrc** [comment]
> /src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp 52 int fuzz = 100

We have a way to find the smallest gap between values and divide that in half. So the numbers will never collide. 
```
 int fuzz = (findMinDiff(joyAdcValues, 6) /2 );
```

We will not hardcode a fuzz of 52 for targets. I think you should take a look at your resistors you are using to get those values.  There are a few examples of other targets and their values.

**Comment by norulers** [comment]
> > /src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp 52 int fuzz = 100
> 
> 我们有一种方法可以找到值之间的最小差距并将其分成两半。所以数字永远不会冲突。
> 
> ```
>  int fuzz = (findMinDiff(joyAdcValues, 6) /2 );
> ```
> 
> 我们不会为目标硬编码 52 的模糊。我认为你应该看看你用来获得这些值的电阻。还有一些其他目标及其值的示例。



> > /src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp 52 int fuzz = 100
> 
> We have a way to find the smallest gap between values and divide that in half. So the numbers will never collide.
> 
> ```
>  int fuzz = (findMinDiff(joyAdcValues, 6) /2 );
> ```
> 
> We will not hardcode a fuzz of 52 for targets. I think you should take a look at your resistors you are using to get those values. There are a few examples of other targets and their values.

52 is 52 line,Send schematic later

**Comment by brandonrc** [comment]
> > > /src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp 52 int fuzz = 100
> > 
> > 
> > We have a way to find the smallest gap between values and divide that in half. So the numbers will never collide.
> > ```
> >  int fuzz = (findMinDiff(joyAdcValues, 6) /2 );
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We will not hardcode a fuzz of 52 for targets. I think you should take a look at your resistors you are using to get those values. There are a few examples of other targets and their values.
> 
> 52 is 52 line ，Send schematic later

So if the line is 52 and you want to set the fuzz to 100 then you will collide. Because you have two button that are `0` and `35`. If you fuzz with 100 then you can have one bleed into the other position (state).

File: src/include/target/AION_NANO_2400_TX.h
Comment (JyeSmith) [inline src/include/target/AION_NANO_2400_TX.h:31]: The e28 does not output 500mW and the max should be changed to 250mW.

**Comment by jurgelenas** [comment]
I have flashed this target to AION Nano module and menu scrolls very fast if you hold direction with joystick even for a little bit. To hop through only one menu screen you need to flick the joystick really fast.

**Comment by schugabe** [comment]
# TX target release checklist

## Misc
- [ ] Device name does not follow naming conventions

## Module update
- [x] uart upload works
- [x] wifi upload works
 
## Display + control
- [ ] RGB LED has correct color: **needs WS2812_IS_GRB**
- [x] logo shows up on startup 
- [ ] joystick control works in all directions: **up does not work, left/right/down jumps multiple items at once if pressed bit long**
  - navigate to refresh rate menu
  - switch rate + save settings
  - open rate menu again and check if value was changed
  - enable wifi update and verify upload works
  - bind to rx via menu
	
## LUA setup
- [x] uart connection works
  - open elrs lua script
  - verify UART connection works
  - test faster baud rates + retest connection
- [ ] bind to rx via lua script works
- [x] check enable wifi works

## RX compability
- [ ] module establishes link to rx of the same vendor (specify rx) **TBD no rx ready to test**
- [x] tlm data is received
- [x] module establishes link to rx of other vendor (specify other rx): EP2
  
## TX power
- [ ] set tx power + verfiy correct output power: **TBD no hw to test**
- [x] set highest tx power and check for temperature issues after 10min: **metal fins get warm but rx rssi stays the same**
- [x] verify rssi dbm value at 1m

File: src/include/target/AION_NANO_2400_TX.h
Comment (AlessandroAU) [reply src/include/target/AION_NANO_2400_TX.h]: turns out this is something different! 
image

Comment (jurgelenas) [reply src/include/target/AION_NANO_2400_TX.h]: Oh cool!

**Comment by AlessandroAU** [comment]
Are you guys able to get BLE working? I am getting a failure and reboot
image

**Comment by JyeSmith** [comment]
I have tested everything.

2 comments.

- Rx wifi range is very short.

- There appears to be 2 version shipping. @norulers can you confirm if the e28 version is shipping to users?  or only for devs?

image

**Comment by maybenikhil** [comment]
> Are you guys able to get BLE working? I am getting a failure and reboot image

BLE Joystick worked fine for me

**Review by JyeSmith — APPROVED** [review]

**Comment by norulers** [comment]
> I have tested everything.
> 
> 3 comments.
> 
> * Rx wifi range is very short.
> * Ipex4 on the rx is terrible.  Especially since there is room for an ipex1!
> * There appears to be 2 version shipping. @norulers can you confirm if the e28 version is shipping to users?  or only for devs?
> 
> image

The delivery time of customized modules is too long. A small batch of samples are E28 modules, and the quantity is expected to be less than 200pcs

**Comment by AlessandroAU** [comment]
Are the E28 units the one going to reviewers? If they are sold to the general public that is bad because not only will they not be able to do the full 500mw but the power levels will be incorrect...

**Comment by JyeSmith** [comment]
@norulers when can we expect a PR for the internal Aion?

**Comment by peregrinus** [comment]
@AlessandroAU not sure if it helps but I purchased one in Australia a few days ago and its not an E28 version. 

image

**Comment by schugabe** [comment]
BLE Joystick also crashes on my module

**Comment by AlessandroAU** [comment]
@schugabe possible an upstream issue? The only difference here is that this is a ESP32-PICO-D4 vs a ESP32

**Comment by schugabe** [comment]
@AlessandroAU with #1322 BLE works on this module

**Review by schugabe — APPROVED** [review]

**Comment by CapnBry** [comment]
Tested this with #1294 and #1322 applied and everything seems to work for me, this seems to be good to go if @AlessandroAU can confirm.

**Comment by pkendall64** [comment]
If merged this will also need #1323

**Review by pkendall64 — APPROVED** [review]

**Review by maybenikhil — APPROVED** [review]

**Comment by maybenikhil** [comment]
Superceded by #1329

**Pull request closed without merging**
